src/codelabs/async-await.md

```diff
@@ -42,6 +42,18 @@ another operation to finish. Here are some common asynchronous operations:
 To perform asynchronous operations in Dart, you can use the `Future` class
 and the `async` and `await` keywords.
 
+Typically, APIs that return `Future` objects
+fall into one of the following categories:
+
+* APIs that are backed by logic running on a different thread. 
+  For example, `File.readAsString` (and most of the other I/O APIs) 
+  call operating system functions from threads on a dedicated thread pool, 
```

Comment:
I wouldn't mention a thread pool, it is not important to the point:
Some futures are waiting for operations running outside of the Dart isolate, other futures are waiting for asynchronous operations inside the Dart isolate.

It's not so much "the thread on which Dart code is executed" as it's the thread that the current isolate is being executed on. Other isolates, also running Dart code, counts as other threads too.

(Those futures completed inside the same isolate are usually completed using microtasks, those from outside the isolate are usually completed by top-level events, but exceptions can occur.)